File: curso/modulo-3/aula-9/ecb.cs
using System.Security.Cryptography;
using System.Text;

var mensagem = "desenvolvedor.io";
RandomNumberGenerator random = RandomNumberGenerator.Create();
byte[] chave = new byte[16];
random.GetBytes(chave);

Console.WriteLine("============== CRIPTOGRAFANDO ==============");

Aes aes = Aes.Create();
aes.Key = chave;
var ciphertext = aes.EncryptEcb(Encoding.UTF8.GetBytes(mensagem), PaddingMode.PKCS7);

Console.WriteLine("Mensagem: {0}", mensagem);
Console.WriteLine("Senha: {0}", Convert.ToHexString(chave));
Console.WriteLine("Cipher: {0}", Convert.ToHexString(ciphertext));
Console.WriteLine();

Console.WriteLine("============== DESCRIPTOGRAFANDO ==============");

Console.WriteLine(Encoding.UTF8.GetString(aes.DecryptEcb(ciphertext, PaddingMode.PKCS7)));

File: curso/modulo-3/aula-9/ecb.py
from os import urandom
from Crypto.Cipher import AES
from Crypto.Util.Padding import pad, unpad
import binascii
from tabulate import tabulate


print('============== CRIPTOGRAFANDO ==============\n')

secretKey = urandom(16)
mensagem = b'desenvolvedor.io'

aes = AES.new(secretKey, AES.MODE_ECB)
cipher = aes.encrypt(pad(mensagem, AES.block_size))


dados = {
    'mensagem': mensagem,
    'senha': binascii.hexlify(secretKey),
    'Cipher': binascii.hexlify(cipher),
}

print(tabulate(dados.items(), tablefmt='fancy_grid'))

print('\n\n============== DESCRIPTOGRAFANDO ==============\n')

plainText = unpad(aes.decrypt(cipher), AES.block_size)
print('Plain text:', plainText.decode())
